beginend.el

```diff
@@ -239,6 +239,13 @@ BEGIN-BODY and END-BODY are two `progn' expressions passed to respectively
   (progn
     (prodigy-last)))
 
+(beginend-define-mode magit-status-mode
+  (progn
+    (re-search-forward "^$")
+    (forward-line))
+  (progn
+    (beginend--goto-nonwhitespace)))
```

Comment:
This gets really crazy for me because magit uses invisible blocks a lot.  These simple searches don't handle it very well.

I think using 

```elisp
(magit-section-backward)
(magit-section-backward)
```

would work best.  Look at the definition of that, it is insanely complicated!  (I think we need to call it twice to get to the beginning of the previous relevant line)